### DISCUSSION TITLE
Keep labelled information for `or` parser

### DISCUSSION THREAD

**Discussion by tfachmann** [zesterer/chumsky#706] [Q&A] [answered]
I am using version `chumsky = { version = "1.0.0-alpha.7", features = ["label"] }`

I am writing a small parser where I optionally want to consume a new line (or an inline comment). For this I am using the `or(...)` parser as follows:

```rust

fn comment<'a, I>() -> impl Parser<'a, I, (), Extra<'a, char>>
where
    I: Input<'a, Token = char, Span = SimpleSpan> + StrInput<'a, char>,
{
    just('#')
        .then(any().and_is(text::newline().not()).repeated())
        .ignored()
        .labelled("comment")
}

fn comment_or_eol<'a, I>() -> impl Parser<'a, I, (), Extra<'a, char>>
where
    I: Input<'a, Token = char, Span = SimpleSpan> + StrInput<'a, char>,
{
    comment()
        .or(text::newline())
        .labelled("comment or end-of-line")
}
```

The parser works as expected, but the labeling does not, and I don't know how to express this differently 😢
When parsing `comment`, I get expected output:

```
Error: found f expected comment
   ╭─[<unknown>:1:1]
   │
 1 │ foo # comment
   │ ┬
   │ ╰── found f expected comment
───╯
```

When parsing `comment_or_eol` however, I get the following:

```
Error: found f expected something else
   ╭─[<unknown>:1:1]
   │
 1 │ foo # comment
   │ ┬
   │ ╰── found f expected something else
───╯
```

Here, I would expect it to say `expected comment or end-of-line` instead of `expected something else`.

**Comment by zesterer** [comment, marked as answer]
Do you get this same issue if you point your project at the latest `main`?

**Reply by tfachmann** [reply]
Thank you for the fast reply, the `main` branch yields the expected behavior.

```toml
chumsky = { git = "https://github.com/zesterer/chumsky", features = ["label"] }
```

Sorry for not trying that out though. I was certain the issue is with me as I'm new to `chumsky`

**Reply by zesterer** [reply]
Good to know. Sorry that this bug made it into a release! Hopefully it won't be long until there's another :)
